### PR TITLE
Implemented and Tested CertifyCreation

### PR DESCRIPTION
**Pull request by matt-tsai**

### CODE REVIEW

File: direct/tpm2/certify_test.go [hunk 1]
```diff
@@ -160,3 +160,100 @@ func TestCertify(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	public := tpm2b.Public{
+		PublicArea: tpmt.Public{
+			Type:    tpm.AlgRSA,
+			NameAlg: tpm.AlgSHA256,
+			ObjectAttributes: tpma.Object{
```
Comment (alexmwu) [inline direct/tpm2/certify_test.go:175]: at some point, we should have similar helper tpma.Object structs that legacy go-tpm has, like `FlagSealDefault` and `FlagSignerDefault` (https://github.com/google/go-tpm/blob/master/tpm2/constants.go#L256-L260). Writing this out can be a bit tedious

File: direct/tpm2/certify_test.go [hunk 2]
```diff
@@ -160,3 +160,100 @@ func TestCertify(t *testing.T) {
...
+				HashAlg: tpm.AlgSHA256,
+			},
+		},
+	}
+
+	certifyCreation := CertifyCreation{
+		SignHandle: AuthHandle{
+			Handle: rspCP.ObjectHandle,
+			Name:   rspCP.Name,
+			Auth:   PasswordAuth([]byte{}),
```
Comment (alexmwu) [inline direct/tpm2/certify_test.go:238]: nit: `PasswordAuth(nil)`

File: direct/tpm2/certify_test.go [hunk 3]
```diff
@@ -160,3 +160,100 @@ func TestCertify(t *testing.T) {
...
+	}
+
+	rspCC, err := certifyCreation.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("Failed to certify creation: %v", err)
+	}
+
+	attName := rspCC.CertifyInfo.AttestationData.Attested.Creation.ObjectName.Buffer
+	pubName := rspCP.Name.Buffer
+	if !cmp.Equal(attName, pubName) {
```
Comment (alexmwu) [inline direct/tpm2/certify_test.go:256]: nit: `if !bytes.Equal(...)`

File: direct/tpm2/certify_test.go [hunk 4]
```diff
@@ -160,3 +160,100 @@ func TestCertify(t *testing.T) {
...
+		t.Fatalf("Attested name: %v does not match returned public key: %v.", attName, pubName)
+	}
```
Comment (alexmwu) [inline direct/tpm2/certify_test.go:258]: can you also validate the signature? that's an important part of CertifyCreation.

**Review by alexmwu — CHANGES_REQUESTED** [review]

File: direct/tpm2/certify_test.go (hunk 1 above)
Comment (matt-tsai) [reply direct/tpm2/certify_test.go:175]: Agreed. I think that should be made in a separate PR though.

File: direct/tpm2/certify_test.go (hunk 2 above)
Comment (matt-tsai) [reply direct/tpm2/certify_test.go:238]: Resolved.

File: direct/tpm2/certify_test.go (hunk 3 above)
Comment (matt-tsai) [reply direct/tpm2/certify_test.go:256]: Resolved.

File: direct/tpm2/certify_test.go (hunk 4 above)
Comment (matt-tsai) [reply direct/tpm2/certify_test.go:258]: Yup. Resolved.

**Review by alexmwu — APPROVED** [review]

**Pull request merged**